crates/vm/levm/src/vm.rs

```diff
@@ -586,9 +594,12 @@ impl VM {
             .balance
             .checked_add(coinbase_fee)
             .ok_or(VMError::BalanceOverflow)?;
-
         self.cache.add_account(&coinbase_address, &coinbase_account);
 
+        if coinbase_fee.is_zero() {
+            self.cache.accounts.remove(&coinbase_address);
+        }
```

Comment:
Can't we add the coinbase fee and after that update the cache only if `coinbase_fee != 0`?

Reply:
Nvm then.